client/src/components/scans/scans.js

```diff
@@ -1,16 +1,187 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+
+import {
+  Alert,
+  Grid,
+  EmptyState,
+  Row,
+  ListView,
+  Modal
+} from 'patternfly-react';
+
+import { getScans } from '../../redux/actions/scansActions';
+
+import ScansToolbar from './scansToolbar';
+import ScansEmptyState from './scansEmptyState';
+import { ScanListItem } from './scanListItem';
+
+class Scans extends React.Component {
+  componentDidMount() {
+    this.props.getScans();
+  }
+
+  matchesFilter(item, filter) {
+    let re = new RegExp(filter.value, 'i');
+
+    switch (filter.field.id) {
+      case 'status':
+        return item.status === filter.value.id;
+      case 'scanType':
+        return item.scan_type === filter.value.id;
+      case 'source':
+        return item.sources.find(source => {
+          return source.name.match(re) !== null;
+        });
+      default:
+        return true;
+    }
+  }
+
+  matchesFilters(item, filters) {
+    let matches = true;
+
+    filters.forEach(filter => {
+      if (!this.matchesFilter(item, filter)) {
+        matches = false;
+        return false;
+      }
+    });
+    return matches;
+  }
+
+  filterScans() {
+    const { scans, activeFilters } = this.props;
+
+    return scans.filter(item => {
+      return this.matchesFilters(item, activeFilters);
+    });
+  }
+
+  sortScans(items) {
+    const { sortType, sortAscending } = this.props;
+
+    let sortId = sortType ? sortType.id : 'name';
+    console.log(sortId);
```

Comment:
Left over ```console.log...```. Also looks like ```name``` is the default filter & sort options... both disappear they're clicked.

![screen shot 2018-01-17 at 1 34 03 pm](https://user-images.githubusercontent.com/3761375/35060041-22a9e056-fb8b-11e7-857f-cbd5cc5e16a5.png)

Reply:
Fixed it up best I could for now. Treating id like name...